tests/integration_test.go

```diff
@@ -0,0 +1,420 @@
+	"net/http"
+	"os"
+	"os/signal"
+	"runtime"
+	"runtime/pprof"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sdoque/mbaigo/components"
+	"github.com/sdoque/mbaigo/forms"
+	"github.com/sdoque/mbaigo/usecases"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+// The most simplest unit asset
+
+const (
+	unitName    string = "randomiser"
+	unitService string = "random"
+)
+
+// Force type check (fulfilling the interface) at compile time
+var _ components.UnitAsset = &uaRandomiser{}
+
+type uaRandomiser struct {
+	Name        string              `json:"-"`
+	Owner       *components.System  `json:"-"`
+	Details     map[string][]string `json:"-"`
+	ServicesMap components.Services `json:"-"`
+	CervicesMap components.Cervices `json:"-"`
+}
+
+// Add required functions to fulfil the UnitAsset interface
+func (ua uaRandomiser) GetName() string                  { return ua.Name }
+func (ua uaRandomiser) GetServices() components.Services { return ua.ServicesMap }
+func (ua uaRandomiser) GetCervices() components.Cervices { return ua.CervicesMap }
+func (ua uaRandomiser) GetDetails() map[string][]string  { return ua.Details }
+func (ua uaRandomiser) Serving(w http.ResponseWriter, r *http.Request, servicePath string) {
+	if servicePath != unitService {
+		http.Error(w, "unknown service path: "+servicePath, http.StatusBadRequest)
+		return
+	}
+	f := forms.SignalA_v1a{
+		Value: rand.Float64(),
+	}
+	b, err := usecases.Pack(f.NewForm(), "application/json")
+	if err != nil {
+		http.Error(w, "error from Pack: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	if _, err := w.Write(b); err != nil {
+		http.Error(w, "error from Write: "+err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func addUATemplate(sys *components.System) {
+	s := &components.Service{
+		Definition: unitService, // The "name" of the service
+		SubPath:    unitService, // Not "allowed" to be changed afterwards
+		Details:    map[string][]string{"key1": {"value1"}},
+		RegPeriod:  60,
+		// NOTE: must start with lower-case, it gets embedded into another sentence in the web API
+		Description: "returns a random float64",
+	}
+	ua := components.UnitAsset(&uaRandomiser{
+		Name:    unitName, // WARN: don't use the system name!! this is an asset!
+		Details: map[string][]string{"key2": {"value2"}},
+		ServicesMap: components.Services{
+			s.SubPath: s,
+		},
+	})
+	sys.UAssets[ua.GetName()] = &ua
+}
+
+func loadUA(ca usecases.ConfigurableAsset, sys *components.System) (components.UnitAsset, func()) {
+	s := ca.Services[0]
+	ua := &uaRandomiser{
+		Name:        ca.Name,
+		Owner:       sys,
+		Details:     ca.Details,
+		ServicesMap: usecases.MakeServiceMap(ca.Services),
+		// Let it consume its own service
+		CervicesMap: components.Cervices{unitService: &components.Cervice{
+			Definition: s.Definition,
+			Details:    s.Details,
+			// Nodes will be filled up by any discovered cervices
+			Nodes: make(map[string][]string, 0),
+		}},
```

Comment:
This is weird and not normal, but I didn't want to mock another service and instead focused on testing the existing unit asset and it's single service.